gpt_index/indices/query/schema.py

```diff
@@ -108,3 +108,21 @@ class QueryConfig(DataClassJsonMixin):
     index_struct_type: str
     query_mode: QueryMode
     query_kwargs: Dict[str, Any] = field(default_factory=dict)
+
+
+@dataclass
+class QueryBundle(DataClassJsonMixin):
+    """
+    Query bundle.
+
+    This dataclass contains the original query string and associated transformations.
+
+    Args:
+        query_str (str): the original user-specified query string.
+            This is currently used by all non embedding-based queries.
+        embedding_strs (list[str]): list of strings used for embedding the query.
```

Comment:
does the user have to specify this? can this be the query_str by default?

Reply:
We continue to support directly passing in just a quer `str` instead of a `QueryBundle` at the main entrypoints (e.g. `BaseGPTIndex` and `ComposableGraph`).

The idea here is that the user would only use `QueryBundle` for customization purpose (and ofc via QueryProcessor in the followup PR.)

Reply:
Revised with your suggestion.